prepare

```diff
@@ -36,7 +36,9 @@ set -ex
 test $VERSION_CODENAME = "jammy" && sudo apt install -y software-properties-common && sudo add-apt-repository -y ppa:v-launchpad-jochen-sprickerhof-de/sbuild
 sudo apt update
 
-sudo DEBIAN_FRONTEND=noninteractive apt install -y --no-install-recommends sbuild mmdebstrap distro-info debian-archive-keyring ccache uidmap curl git-lfs qemu-user-static arch-test
+sudo DEBIAN_FRONTEND=noninteractive apt install -y --no-install-recommends sbuild mmdebstrap distro-info debian-archive-keyring ccache uidmap curl git-lfs
+
+test $DEB_ARCH != $(dpkg --print-architecture) && sudo apt install -y qemu-user-static arch-test binfmt-support && sudo update-binfmts --enable
```

Comment:
binfmt-support is only needed in jammy. Can you add a conditional logic?

Reply:
Why is the support not needed for other distributions? Is this related to the `ppa:v-launchpad-jochen-sprickerhof-de/sbuild`?

Reply:
No, it is done by the package automatically. Actually I think it should work without it on jammy as well. Any reason you added it?

Reply:
Without this, the CI failed with:
```
W: binfmt_misc not found in /proc/mounts -- not mounted?
E: arm64 can neither be executed natively nor via qemu user emulation with binfmt_misc
```

Reply:
On which Ubuntu version is this? Can you provide a link to the log?

Reply:
This is in an `ubuntu:22.04` Docker container: https://github.com/christianrauch/camera-ros-deb-builder/actions/runs/8790256266/job/24121874782 and should be reproducible locally with the same Docker image.

Reply:
Docker is not the right tool to check as this as it is kernel interface and namespacing binfmt_misc was only added in Linux 6.7 and Github runners are VMs. You can test it like this:
```shell
debvm-create -z 2G --release noble -- --keyring=/usr/share/keyrings/ubuntu-archive-keyring.gpg --components=main,universe --include=e2fsprogs
debvm-run
# inside debvm
apt install qemu-user-static
ls /proc/sys/fs/binfmt_misc/
```
Which works fine for me in jammy and noble.

If you want to make it work on Docker based systems you may try `arch-test` and in case add some workarounds but note that this is rather brittle, see for example all mentions of `binfmt` in `mmdebstrap` here and elsewhere: https://sources.debian.org/src/mmdebstrap/1.4.3-6/mmdebstrap/#L5151

Reply:
> Docker is not the right tool to check as this as it is kernel interface and namespacing binfmt_misc was only added in Linux 6.7 and Github runners are VMs.

This works in a privileged Docker container:
```sh
docker run -it --rm --privileged ubuntu:22.04 bash
```

Installing `qemu-user-static` and running `update-binfmts --enable`:
```sh
apt update && apt -y install qemu-user-static
update-binfmts --enable
grep binfmt_misc /proc/mounts
```
will show you the mounted `binfmt_misc` file system:
```
binfmt_misc /proc/sys/fs/binfmt_misc binfmt_misc rw,nosuid,nodev,noexec,relatime 0 0
```

Without `update-binfmts --enable` the `binfmt_misc` file system will no be mounted.

> Edit: looking at your log, why do you set a `container:` in there?

I am running the action inside a Docker container to have this better reproducible locally and mainly to be independent from the runner version. The former lets you discover missing dependencies (such as the missing `software-properties-common` package) and the latter lets you use newer Ubuntu versions (and sbuild etc.) than what is available via the official runners.

Reply:
> This works in a privileged Docker container:

As written, this depends on your host system and I am fine to add this conditioned on an extra test.